himan-scripts/cloud-consensus.lua

```diff
@@ -0,0 +1,223 @@
+    ftime = forecast_time(raw_time(test_time), raw_time("2025-04-04 06:00:00"))
+    return ftime
+  end
+
+  local vire_hour = current_time:GetOriginDateTime():String('%H')
+  local producer_id = producer:GetId()
+
+  if producer_id == 4 or producer_id == 260 then
+    adjust_hours = -4
+  elseif (vire_hour == '07' or vire_hour == '19') and (producer_id == 131 or producer_id == 242) then
+      adjust_hours = -7
+  elseif (vire_hour == '13' or vire_hour == '01') and (producer_id == 131 or producer_id == 242) then
+      adjust_hours = -13
+  end
+
+  if adjust_hours then
+    current_time:GetOriginDateTime():Adjust(HPTimeResolution.kHourResolution, adjust_hours)
+  end
+  
+  ftime = forecast_time(current_time:GetOriginDateTime(), current_time:GetValidDateTime())
+  current_time:GetOriginDateTime():Adjust(HPTimeResolution.kHourResolution, -adjust_hours) -- reset to original time
+  return ftime
+end
+
+
+
+function get_param(producer, ftype, param1, param2, param3)
+
+  local ftime = get_time(producer)
+
+  
+  local o = {forecast_time = ftime,
+  level = level(HPLevelType.kHeight, 0),
+  param = param1,
+  forecast_type = ftype,
+  time_interpolation = true,
+  time_interpolation_search_step = time_duration("01:00:00"),
+  producer = producer}
+  local param1 = luatool:FetchWithArgs(o)
+
+  o.param = param2
+  local param2 = luatool:FetchWithArgs(o)
+
+  o.param = param3
+  local param3 = luatool:FetchWithArgs(o)
+  return param1, param2, param3
+end
+
+function compute_levels(step)
+  local mid_level = 50 + step^(1.5) / 500
+  local dev_factor_low = 1.2 + step ^(1.5) / 10000
+  local dev_factor_mid = 1.1 + step^ (1.5) / 12500
+  local dev_factor_high = 1 + step^ (1.5) / 15000
+  return mid_level, dev_factor_low, dev_factor_mid, dev_factor_high
+end
+
+function compute_weights(step)
+  local ratio = math.min(0.0001175 * step ^ 2 + 0.35, 1) 
+  local wt_meps = (1 - ratio) * (1/4)
+  local wt_ens_meps = (1 - ratio) * (3/4)
+  local wt_ec = ratio * (1 - 0.1818 * step^(0.25867))
+  local wt_ens_ec = ratio - wt_ec
+  return wt_meps, wt_ens_meps, wt_ec, wt_ens_ec
+end
+
+
+local step = current_time:GetStep():Hours()
+
+local wt_meps, wt_ens_meps, wt_ec, wt_ens_ec = compute_weights(step)
+local mid_level, dev_factor_low, dev_factor_mid, dev_factor_high = compute_levels(step)
+
+local mean, stDev = nil, nil
+local cl, cm, ch = {}, {}, {} 
+
+
+local disable_meps = nil
+if configuration:Exists("disable_meps") then 
+  disable_meps = ParseBoolean(configuration:GetValue("disable_meps"))
+end
+
+meps_time = get_time(producer(4, "MEPS"))
+meps_step = tonumber(meps_time:GetStep():Hours())
+
+if disable_meps or meps_step > 66 then
+  logger:Info("Only using EC data")
+  
+  NL_EC, NM_EC, NH_EC, NL_MEAN_EC, NM_MEAN_EC, NH_MEAN_EC, NL_STD_EC, NM_STD_EC, NH_STD_EC = get_data(producer(131, "ECG"), producer(242, "ECM_PROB"), forecast_type(HPForecastType.kDeterministic))
+  
+  if not NL_EC or not NM_EC or not NH_EC then
+    logger:Warning("Some data not found")
```

Comment:
if NL_EC is not found, the loop below won't execute and the whole data will not be formed. I guess this is what we want? In this case might make it more explicit, ie logging and Error (not Warning) and exiting the script at this point.